otherlibs/unix/socketaddr.h

```diff
@@ -64,16 +64,25 @@ typedef int socklen_param_type;
 extern "C" {
 #endif
 
-extern void get_sockaddr (value mladdr,
-                          union sock_addr_union * addr /*out*/,
-                          socklen_param_type * addr_len /*out*/);
-extern value alloc_sockaddr (union sock_addr_union * addr /*in*/,
-                      socklen_param_type addr_len, int close_on_error);
-extern value alloc_inet_addr (struct in_addr * inaddr);
+/* Compatibility definitions for the pre-5.0 names of these functions */
+#ifndef CAML_BUILDING_UNIX
+#define get_sockaddr caml_unix_get_sockaddr
+#define alloc_sockaddr caml_unix_alloc_sockaddr
+#define alloc_inet_addr caml_unix_alloc_inet_addr
+#define alloc_inet6_addr caml_unix_alloc_inet6_addr
```

Comment:
I'd move this one inside the `#ifdef HAS_IPV6` on line 84.

Reply:
Good idea, yes!

```diff
diff --git a/otherlibs/unix/socketaddr.h b/otherlibs/unix/socketaddr.h
index 66e7b02aac..768669d0cc 100644
--- a/otherlibs/unix/socketaddr.h
+++ b/otherlibs/unix/socketaddr.h
@@ -69,7 +69,6 @@ extern "C" {
 #define get_sockaddr caml_unix_get_sockaddr
 #define alloc_sockaddr caml_unix_alloc_sockaddr
 #define alloc_inet_addr caml_unix_alloc_inet_addr
-#define alloc_inet6_addr caml_unix_alloc_inet6_addr
 #endif /* CAML_BUILDING_UNIX */

 extern void caml_unix_get_sockaddr (value mladdr,
@@ -84,7 +83,12 @@ extern value caml_unix_alloc_inet_addr (struct in_addr * inaddr);
 #ifdef HAS_IPV6
 extern value caml_unix_alloc_inet6_addr (struct in6_addr * inaddr);
 #define GET_INET6_ADDR(v) (*((struct in6_addr *) (v)))
-#endif
+
+/* Compatibility definition for the pre-5.0 name of this function */
+#ifndef CAML_BUILDING_UNIX
+#define alloc_inet6_addr caml_unix_alloc_inet6_addr
+#endif /* CAML_BUILDING_UNIX */
+#endif /* HAS_IPV6 */

 #ifdef __cplusplus
 }
```